prereise/gather/data/remap_ba_area/remap_ba_area.py

```diff
@@ -0,0 +1,294 @@
+    caiso_subregions.rename({"NAME": "BA"}, axis="columns", inplace=True)
+    # Tag the existing geometries with their true CRS, then re-project
+    caiso_subregions = caiso_subregions.set_crs(3857, allow_override=True)
+    caiso_subregions = caiso_subregions.to_crs(4269)
+    california_joined = pd.concat([california.reset_index(), caiso_subregions])
+    # We simultaneously concatenate and dissolve
+    full_ba_map = (
+        pd.concat([ba_dissolved.reset_index(), california_joined])
+        .dissolve(by="BA")["geometry"]
+        .to_frame()
+    )
+    # Subtract the CAISO Valley Electric Association geometry from the Nevada Power geometry
+    full_ba_map.loc["NEVP", "geometry"] = full_ba_map.loc[
+        "NEVP", "geometry"
+    ].difference(full_ba_map.loc["CISO-VEA", "geometry"])
+
+    nyiso = get_nyiso_area(crs=full_ba_map.crs)
+    full_ba_map = full_ba_map.append(nyiso)
+    # add ISONE
+    full_ba_map = full_ba_map.drop("ISNE")
+    isone_df = gpd.read_file(get_full_filename("isone.zip"))[
+        ["STUSPS", "geometry"]
+    ].rename(columns={"STUSPS": "BA"})
+    isone_df = isone_df.assign(BA=isone_df["BA"].apply(lambda t: "ISONE-" + t))
+    full_ba_map = pd.concat([full_ba_map, isone_df.set_index("BA")])
+    # assign the four utility territories into existing CA/subregion
+    # rename TVA_LGEE to TVA, to match the source name
+    extra_map = {
+        "NV Energy": "CISO-PGAE",
+        "SCL_TPWR": "BPAT",
+        "TID": "CISO-PGAE",
+        "WAPA": "CISO-SCE",
+        "TVA_LGEE": "TVA",
+    }
+    full_ba_map = full_ba_map.reset_index().replace({"BA": extra_map}).set_index("BA")
+    full_ba_map = full_ba_map.dissolve(by="BA")
+    # matched source names
+    ca_name_df["fraction"] = 1
+    # unmatched source names
+    crs = full_ba_map.crs
+    control_area_df = (
+        gpd.read_file(get_full_filename("Control_Areas.zip")).to_crs(crs).reset_index()
+    )
+    ca_name_missing_df = ca_name_missing_df.rename(
+        columns={"missing": "NAME"}
+    ).set_index("NAME")
+    control_area_df = (
+        control_area_df[["NAME", "geometry"]]
+        .set_index("NAME")
+        .join(ca_name_missing_df)
+        .dropna()
+    )
+    cf = control_area_df.reset_index()
+    cf["area"] = cf["geometry"].to_crs(CRS("epsg:3857")).area
+    cf_mapped = cf.overlay(
+        full_ba_map.reset_index(),
+        how="intersection",
+        keep_geom_type=False,
+        make_valid=True,
+    )
+    cf_mapped["intersection_area"] = cf_mapped["geometry"].to_crs(CRS("epsg:3857")).area
+    cf_mapped["fraction"] = cf_mapped["intersection_area"] / cf_mapped["area"]
+    cf_mapped = cf_mapped[cf_mapped["fraction"] > min_fraction_to_keep]
+    cf_mapped = cf_mapped.set_index("ca_source")
+    source_ca_df = pd.concat(
+        [
+            ca_name_df.set_index(src).rename(columns={dst: "BA"}),
+            cf_mapped[["BA", "fraction"]],
+        ]
+    )
+    source_ca_df.to_csv(get_full_filename("source_map_to_ca.csv"))
```

Comment:
What's the purpose of generating this CSV? Is this redundant with the logic in #271?

Reply:
Looks it is a duplicate. I'll remove these codes, and only keep the codes to generate the map.